### ISSUE TITLE
[Java] False positive: LawOfDemeter must permit package-private access. 

### ISSUE THREAD

**Issue by Fernal73** [pmd/pmd#2182]
**Affects PMD Version:** 
6.20.0
**Rule:**
LawOfDemeter.
**Description:**
Direct access of fields is flagged as static access. Direct access of fields should be permitted when allowed for package-private.

**Code Sample demonstrating the issue:**

```java
import java.time.LocalDate;

public class Person {
  LocalDate birthday;

  public static int compareByAge(Person a, Person b) {
    return a.birthday.compareTo(b.birthday);
  }
}

```

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

**Comment by jsotuyod** [comment]
This is fixed in PMD 7.0.0-rc1. It won't be backported to PMD 6.
